[Joseph Aedan Marcus] iP

Josephaedan opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's

* Text-based
* Easy to learn
* ~~Fast~~ SUPER FAST to use

All you need to do is

1. Download it from [here](https://github.com/josephaedan/ip/releases)
2. Save it into a folder
3. Add your tasks
4. Let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [ ] Create tasks from 3 categories: Todo, Deadline, Event
- [ ] Mark and unmark any task as done
- [ ] Filter tasks by searching for keywords
- [ ] Reminders (coming soon)
# 
If you're a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
public static void main(String[] args) {
    Application.launch(Main.class, args);
}
```

Commit:
Add Gradle support

Josephaedan committed:
Level-1

Josephaedan committed:
Level-1

Josephaedan committed:
level 2

Josephaedan committed:
Level 3

Josephaedan committed:
level 4 changes

Josephaedan committed:
automated testing

Josephaedan committed:
Level 5 changes

Josephaedan committed:
Level 6 changes

Josephaedan committed:
enums

Josephaedan committed:
Level-7 changes

Josephaedan committed:
Update Duke.java

Josephaedan committed:
Level 8 LocalDate format

Josephaedan committed:
Merge pull request #1 from Josephaedan/branch-Level-7

Level-7 changes

Josephaedan committed:
Merge branch 'master' into branch-Level-8

Josephaedan committed:
Merge pull request #2 from Josephaedan/branch-Level-8

Branch level 8

Josephaedan committed:
Merge branch 'master' into add-gradle-support

Josephaedan committed:
Merge pull request #3 from Josephaedan/add-gradle-support

Add gradle support

nicolelim02 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,162 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.time.LocalDate;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.io.FileWriter;
+
```
Perhaps you can reorder the import statements such that it complies with the coding standard

nicolelim02 commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,13 @@
+public class DukeException extends Exception {
+    String message;
```
Should this be set as private?

nicolelim02 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,162 @@
+            try {
+                String input = scanner.nextLine();
+                String[] inputArray = input.split(" ", 2);
+                Commands command = Commands.parseCommand(inputArray[0])
+                        .orElseThrow(() -> new DukeException("Sorry, I don't understand you. Please try again."));
```
Perhaps this line could be rewritten as:
```
Commands.parseCommand(inputArray[0])
          .orElseThrow(() -> new DukeException("Sorry, I don't understand you. Please try again."));
```

nicolelim02 reviewed (commented):
I find your code easy to read in general. Keep up the good work!

yellow-294 commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,27 @@
+        this.text = command;
+    }
+
+    public static Optional<Commands> parseCommand(String command) {
+        for (Commands c : Commands.values()) {
+            if (c.text.equals(command)) {
+                return Optional.of(c);
+            }
+        }
+        return Optional.empty();
+    }
+}
```
Interesting use of Optional here 👍

yellow-294 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,162 @@
+                    Deadline deadline = new Deadline(description, date, time);
+                    list.add(deadline);
+
+                    printWithLineBreak(
+                            "Got it. I've added this task:\n" + deadline + "\nNow you have " + list.size()
```
Is it possible to make a Ui class to handle all your display messages?

yellow-294 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,162 @@
+        Scanner scanner = new Scanner(System.in);
+
+        // list of items
+        ArrayList<Task> list = new ArrayList<>();
```
Is it possible to create a TaskList class to encapsulate ur Tasks?

yellow-294 reviewed (commented):
Overall good, functional code.

However, I feel that you could have made separate classes to handle the parsing of user inputs and display messages

Josephaedan committed:
A-More OOP

Josephaedan committed:
A-MoreOOP

Josephaedan committed:
no message

Josephaedan committed:
A-JUnit tests and change UI printing to string

Josephaedan committed:
Junit

Josephaedan committed:
A-Jar

Josephaedan committed:
A-Javadoc

Josephaedan committed:
Merge pull request #5 from Josephaedan/branch-A-JavaDoc

A-Javadoc

Josephaedan committed:
Update UiTest.java

Rename test functions in UiTest according to Java Coding standard

Current methods are named improperly according to coding standard.

Leaving the method names as is would make the code hard to read and comprehend.

Let's rename the functions according to the coding standard.

This way, it is clear to anybody reading our test function what our test function does.

Josephaedan committed:
Merge pull request #6 from Josephaedan/branch-A-CodingStandard

Update UiTest.java

Josephaedan committed:
Add Find functionality

Currently, there is no way to find tasks that a user is looking for.

This makes filtering down tasks or tracking tasks difficult for users, especially in the future when they may have multiple tasks saved.

Let's add a find functionality that will filter their tasks based on the search keyword that they provide.

This helps users to easily find tasks that contain their keywords for easier accessibility and filtering.

Josephaedan committed:
Merge pull request #7 from Josephaedan/branch-Level-9

Add Find functionality

Josephaedan committed:
Added GUI with JavaFX

Currently, Duke does not have a GUI for interactive user experiences.

This severely hampers the usability and appeal for laymen users who do not appreciate the CLI interface.

Let's add a GUI here to allow Duke to run as a chatbot-like application.

This would help aid the user experience and improve the readability of Duke.

Josephaedan committed:
Merge pull request #8 from Josephaedan/branch-Level-10

Added GUI with JavaFX

Josephaedan committed:
Updated Packages

Josephaedan committed:
Added assertions to logic in main functions in Duke

Josephaedan committed:
Misc Code Quality changes

Josephaedan committed:
Merge pull request #9 from Josephaedan/branch-A-Assertions

Added assertions to logic in main functions in Duke

Josephaedan committed:
Extension for Detection of duplicates

Josephaedan committed:
Merge pull request #11 from Josephaedan/branch-C-DetectDuplicates

Extension for Detection of duplicates

Josephaedan committed:
Update Duke.java

Josephaedan committed:
Merge branch 'master' into branch-A-CodeQuality

Josephaedan committed:
Merge pull request #10 from Josephaedan/branch-A-CodeQuality

Misc Code Quality changes

Josephaedan committed:
Release User Guide

Josephaedan committed:
Changed DaDuke.png

Josephaedan committed:
Delete UI.png

Josephaedan committed:
Create Ui.png

Josephaedan committed:
Change to Rick
